message/event.go

```diff
@@ -0,0 +1,123 @@
+package message
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/xmidt-org/wrp-go/v3"
+)
+
+const (
+	BootTimeKey = "/boot-time"
+)
+
+var (
+	ErrParseDeviceID    = errors.New("error getting device ID from event")
+	ErrBirthdateParse   = errors.New("unable to parse birthdate from payload")
+	ErrBootTimeParse    = errors.New("unable to parse boot-time")
+	ErrBootTimeNotFound = errors.New("boot-time not found")
+)
+
+// Event is the struct that contains the wrp.Message fields along with the birthdate
+// that is parsed from the payload.
+type Event struct {
+	MsgType         int               `json:"msg_type"`
+	Source          string            `json:"source"`
+	Destination     string            `json:"dest,omitempty"`
+	TransactionUUID string            `json:"transaction_uuid,omitempty"`
+	ContentType     string            `json:"content_type,omitempty"`
+	Metadata        map[string]string `json:"metadata"`
+	Payload         string            `json:"payload,omitempty"`
+	Birthdate       int64             `json:"birth_date"`
+	PartnerIDs      []string          `json:"partner_ids,omitempty"`
+}
+
+// NewEvent creates an Event from a wrp.Message and also parses the Birthdate from the
+// message payload, setting the birthdate to the defaultBirthdate if it cannot be parsed.
+func NewEvent(msg wrp.Message) (Event, error) {
+	var err error
+	event := Event{
+		MsgType:         int(msg.MessageType()),
+		Source:          msg.Source,
+		Destination:     msg.Destination,
+		TransactionUUID: msg.TransactionUUID,
+		ContentType:     msg.ContentType,
+		Metadata:        msg.Metadata,
+		Payload:         string(msg.Payload),
+		PartnerIDs:      msg.PartnerIDs,
+	}
+
+	if birthdate, ok := getBirthDate(msg.Payload); ok {
+		event.Birthdate = birthdate.UnixNano()
+	} else {
+		err = ErrBirthdateParse
+	}
+
+	return event, err
+}
+
+// Checks a map for a specific key, allowing for keys with or without forward-slash
+func (e Event) GetMetadataValue(key string) (string, bool) {
+	value, found := e.Metadata[key]
+	if !found {
+		value, found = e.Metadata[strings.Trim(key, "/")]
+	}
+
+	return value, found
+}
+
+// BootTime parses the boot-time from an event, returning an
+// error if the boot-time doens't exist or cannot be parsed.
+func (e Event) BootTime() (int64, error) {
+	var bootTime int64
+	var err error
+	bootTimeStr, ok := e.GetMetadataValue(BootTimeKey)
+
+	if ok {
+		bootTime, err = strconv.ParseInt(bootTimeStr, 10, 64)
+		if err != nil {
+			return 0, fmt.Errorf("%w: %v", ErrBootTimeParse, err)
+		}
+	} else {
+		err = ErrBootTimeNotFound
+	}
+
+	return bootTime, err
+}
```

Comment:
I'm pretty sure you just moved this function as a part of this pr, but since I'm here, we can improve it a bit! 🙂 Most go code follows a "fail fast"/"return early" pattern.  The pattern allows for Go code to be [indented as little as possible](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow), but it also helps each function have a similar flow and can aid in readability and maintainability long term.

Here is my suggestion for how to structure this function:
```
func (e Event) BootTime() (int64, error) {
	bootTimeStr, ok := e.GetMetadataValue(BootTimeKey)
	if !ok {
		return 0, ErrBootTimeNotFound
	}
	bootTime, err := strconv.ParseInt(bootTimeStr, 10, 64)
	if err != nil {
		return 0, fmt.Errorf("%w: %v", ErrBootTimeParse, err)
	}
	return bootTime, nil
}
```

This allows everything to be more flat and removes the need for declaring variables at the beginning of the function, since they are being created and returned in the same scope.